File: hackerrank/One-Month-Preparation-Kit/Week-1/plus-minus/plus-minus.dart
void main() {
  List<int> arr = [-4, 3, -9, 0, 4, 1];
  plusMinus(arr);
}

void plusMinus(List<int> arr) {
  int positive = 0;
  int minus = 0;
  int zero = 0;

  for (int i = 0; i < arr.length; i++) {
    if (arr[i] > 0) {
      positive++;
    } else if (arr[i] < 0) {
      minus++;
    } else {
      zero++;
    }
  }
  print((positive / arr.length).toDouble());
  print((minus / arr.length).toDouble());
  print((zero / arr.length).toDouble());
}

File: hackerrank/One-Month-Preparation-Kit/Week-1/plus-minus/plus-minus.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function (inputStdin) {
  inputString += inputStdin;
});

process.stdin.on('end', function () {
  inputString = inputString.split('\n');

  main();
});

function readLine() {
  return inputString[currentLine++];
}

/*
 * Complete the 'plusMinus' function below.
 *
 * The function accepts INTEGER_ARRAY arr as parameter.
 */

function plusMinus(arr) {
  let positive = 0;
  let minus = 0;
  let zero = 0;
  for (let index = 0; index < arr.length; index++) {
    const element = arr[index];
    if (element > 0) {
      positive++;
    } else if (element < 0) {
      minus++;
    } else {
      zero++;
    }
  }
  console.log(positive / arr.length)
  console.log(minus / arr.length)
  console.log(zero / arr.length)

}

function main() {
  const n = parseInt(readLine().trim(), 10);

  const arr = readLine().replace(/\s+$/g, '').split(' ').map(arrTemp => parseInt(arrTemp, 10));

  plusMinus(arr);
}
